### PR TITLE
Add optional encryption_at_host_enabled value

### PR DESCRIPTION
**Pull request by kiebrew**
# [Issue-id](https://github.com/aztfmod/terraform-azurerm-caf/issues/ISSUE-ID-GOES-HERE)

## PR Checklist

---

- [ ] I have added example(s) inside the [./examples/] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Description
Simple change which adds the optional encyrption_at_host_enabled parameter to the existing Azure Windows VM Module. This is now supported within the azurerm terraform provider: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/windows_virtual_machine#encryption_at_host_enabled

## Does this introduce a breaking change

- [ ] YES
- [x] NO

### CODE REVIEW

File: modules/compute/virtual_machine/vm_windows.tf [hunk 1]
```diff
@@ -53,6 +53,7 @@ resource "azurerm_windows_virtual_machine" "vm" {
   availability_set_id          = can(each.value.availability_set_key) || can(each.value.availability_set.key) ? var.availability_sets[try(var.client_config.landingzone_key, each.value.availability_set.lz_key)][try(each.value.availability_set_key, each.value.availability_set.key)].id : try(each.value.availability_set.id, each.value.availability_set_id, null)
   computer_name                = azurecaf_name.windows_computer_name[each.key].result
   enable_automatic_updates     = try(each.value.enable_automatic_updates, null)
+  encryption_at_host_enabled   = try(each.encryption_at_host_enabled, null)
```
Comment (arnaudlh) [inline modules/compute/virtual_machine/vm_windows.tf:56]:
```suggestion
  encryption_at_host_enabled   = try(each.value.encryption_at_host_enabled, null)  
```

**Review by arnaudlh — CHANGES_REQUESTED** [review]
Can you please review as per suggestion, and add the same for Linux VM if applicable?

**Review by kiebrew — COMMENTED** [review]
Updated using proper syntax and added setting to Linux VM, which is also supported within the azurerm_linux_virtual_machine provider:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#encryption_at_host_enabled

**Review by arnaudlh — APPROVED** [review]
LGTM, thanks for the update!

**Pull request merged**